libraries/AP_Param/AP_Param.h

```diff
@@ -542,11 +542,16 @@ class AP_ParamT : public AP_Param
             set(v);
         }
     }
-    
+
     /// Value setter - set value, tell GCS
     ///
     void set_and_notify(const T &v) {
+// We do want to compare each value, even floats, since it being the same here
+// is the result of previously setting it.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wfloat-equal"
         if (v != _value) {
+#pragma GCC diagnostic pop
```

Comment:
So... the way it is now, AP_Math.h includes AP_Param.h. We can't include AP_Math.h here and trying to move the dependency proved difficult for things like `is_zero()` that must know the definition of AP_Float.